2_process/src/add_dynamic_attr.R

```diff
@@ -0,0 +1,796 @@
+    df <- cbind(df, compute_lagged_attrs(dyn_df = df_lag, 
+                                         lag_table = lag_table, 
+                                         lag_attr_name = 'Land',
+                                         attr_name = '',
+                                         start_date = start_date) %>%
+                  select(-Date, -seg)
+    )
+  }
+    
+  
+  #gridMET data
+  if (!is.null(gridMET)){
+    #Reformat columns
+    gridMET <- filter(gridMET, time <= end_date, time >= start_date) %>%
+      #format for new variable is NAME_lag, lag = 0 for these variables
+      rename(Date = time, seg = PRMS_segid, tmmx_0 = tmmx, tmmn_0 = tmmn, 
+             pr_0 = pr, srad_0 = srad, 
+             vs_0 = vs, rmax_0 = rmax, rmin_0 = rmin, sph_0 = sph)
+    
+    #add lagged dynamic information
+    gridMET <- compute_lagged_attrs_from_dynamic(dyn_df = gridMET, 
+                                                 lag_table = lag_table, 
+                                                 lag_attr_name = 'Met',
+                                                 start_date = start_date)
+    
+    #Join to df
+    df <- left_join(df, gridMET, by = c('seg', 'Date'))
+  }
+  
+  #Monthly baseflow data
+  if (!is.null(baseflow)){
+    #Add year and month columns (to be dropped after joining)
+    df$Year <- year(df$Date)
+    df$Month <- month(df$Date)
+    #Remove the leading 0 for Month in baseflow
+    baseflow$Month <- as.numeric(baseflow$Month)
+    #format for new variable is NAME_lag, lag = 0 for these variables
+    baseflow <- rename(baseflow, 
+           mean_natl_baseflow_cfs_0 = mean_natl_baseflow_cfs,
+           med_natl_baseflow_cfs_0 = med_natl_baseflow_cfs,
+           p10_natl_baseflow_cfs_0 = p10_natl_baseflow_cfs,
+           p90_natl_baseflow_cfs_0 = p90_natl_baseflow_cfs)
+    #Join data to all segs by year and month
+    df <- left_join(df, baseflow, by = c('seg' = 'PRMS_segid', 'Year', 'Month')) %>%
+      select(-Year, -Month)
+    
+    #add lagged dynamic information
+    #for baseflow, the start_date is the earliest available date, so this is 
+    #the df_lag.
+    df_lag <- df %>% 
+      select(Date, seg, contains('baseflow'))
+    
+    #Columns to remove before joining because they're already in df
+    col_select <- colnames(df_lag %>% select(contains('baseflow')))
+    
+    #Use the lag function to compute the desired attributes
+    df <- cbind(df, compute_lagged_attrs_from_dynamic(dyn_df = df_lag, 
+                                                      lag_table = lag_table, 
+                                                      lag_attr_name = 'Baseflow',
+                                                      start_date = start_date) %>%
+                  select(-Date, -seg, -all_of(col_select))
+    )
+  }
+
+  return(df)
+}
+
+get_four_digit_year <- function(two_digit_yr){
+  #' @description computes the 4 digit year from two digit year
```

Comment:
It might be useful to add to this function description that `get_four_digit_year()` is specific to the `HDENS` data because of the range in year included in that dataset. For example, if an attribute dataset included data from 1930, this function would not work as intended.